webdav3/client.py

```diff
@@ -849,6 +853,38 @@ class WebDavXmlUtils:
     def __init__(self):
         pass
 
+    @staticmethod
+    def parse_get_list_info_response(content):
+        """Parses of response content XML from WebDAV server and extract file and directory infos
+
+        :param content: the XML content of HTTP response from WebDAV server for getting list of files by remote path.
+        :return: list of extracted file or directory infos.
+        """
+        try:
+            tree = etree.fromstring(content)
+            infos = []
+            for response in tree.findall(".//{DAV:}response"):
+                href_el = next(iter(response.findall(".//{DAV:}href")), None)
+                if href_el is None:
+                    continue
+
+                info = dict()
+                is_dir = len(response.findall(".//{DAV:}collection")) > 0
+                find_attributes = {
```

Comment:
@huangganggui it duplicates code in line 957, lets extract processing of response for getting info in separated method and cover it by tests